MovieApp/MoviePictureCell.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  MovieCatergoyCell.swift
+//  MovieApp
+//
+//  Created by endava-bootcamp on 02.05.2023..
+//
+
+import Foundation
+import UIKit
+import PureLayout
+import MovieAppData
+
+class MoviePictureCell: UICollectionViewCell {
+    static let identifier = "MoviePictureCell"
+    
+    var navigationController: UINavigationController!    
+    var img : UIImage!
+    var imgView: UIImageView!
+    var likeButton : UIButton!
+    var id : Int!
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        buildViews()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func buildViews() {
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+        self.autoSetDimensions(to: CGSize(width: self.frame.width, height: self.frame.height))
```

Comment:
At this point in time cell might not have a defined width and height

You can try to extract the needed height and width values into a constant that you will use wherever you need to set it explicitly. That way you have a single source of truth across the views of the cell dimensions

also, this should be moved to the `defineLayoutForViews`